src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -1799,6 +1797,360 @@ static class D
             );
     }
 
+    [Fact]
+    public void InterceptableGeneric_05()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            C.Usage(1);
+            C.Usage(2);
+
+            class C
+            {
+                public static void InterceptableMethod<T1>(T1 t) => throw null!;
+
+                public static void Usage<T2>(T2 t)
+                {
+                    C.InterceptableMethod(t);
+                    C.InterceptableMethod<T2>(t);
+                    C.InterceptableMethod<object>(t);
+                }
+            }
+
+            static class D
+            {
+                [InterceptsLocation("Program.cs", 13, 11)]
+                [InterceptsLocation("Program.cs", 14, 11)]
+                [InterceptsLocation("Program.cs", 15, 11)]
+                public static void Interceptor1<T>(T t) { Console.Write(t); }
+            }
+            """;
+        var verifier = CompileAndVerify(new[] { (source, "Program.cs"), s_attributesSource }, parseOptions: RegularWithInterceptors, expectedOutput: "111222");
+        verifier.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void InterceptableGeneric_06()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            class C
+            {
+                public static void InterceptableMethod<T1>(T1 t) => throw null!;
+
+                public static void Usage()
+                {
+                    C.InterceptableMethod("abc");
+                }
+            }
+
+            static class D
+            {
+                [InterceptsLocation("Program.cs", 10, 11)] // 1
+                public static void Interceptor1<T>(T t) where T : struct => throw null!;
```

Comment:
>where T : struct [](http://example.com/codeflow?start=56&length=16)

Let's also test with constraints relative to nullability (a nullable or non-nullable reference type, or the `notnull` constraint)